app/controllers/nonsipp/membercontributions/RemoveMemberContributionController.scala

```diff
@@ -0,0 +1,136 @@
+import forms.YesNoPageFormProvider
+import models.{Money, NormalMode}
+import models.SchemeId.Srn
+import navigation.Navigator
+import pages.nonsipp.membercontributions.{RemoveMemberContributionPage, TotalMemberContributionPage}
+import pages.nonsipp.memberdetails.MemberDetailsPage
+import play.api.data.Form
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.SaveService
+import viewmodels.DisplayMessage.Message
+import viewmodels.models.{FormPageViewModel, YesNoPageViewModel}
+import views.html.YesNoPageView
+import viewmodels.implicits._
+
+import javax.inject.{Inject, Named}
+import scala.concurrent.{ExecutionContext, Future}
+
+class RemoveMemberContributionController @Inject()(
+  override val messagesApi: MessagesApi,
+  @Named("non-sipp") navigator: Navigator,
+  identifyAndRequireData: IdentifyAndRequireData,
+  formProvider: YesNoPageFormProvider,
+  saveService: SaveService,
+  val controllerComponents: MessagesControllerComponents,
+  view: YesNoPageView
+)(implicit ec: ExecutionContext)
+    extends PSRController
+    with I18nSupport {
+
+  private val form = RemoveMemberContributionController.form(formProvider)
+
+  def onPageLoad(srn: Srn, memberIndex: Max300, index: Max50): Action[AnyContent] =
+    identifyAndRequireData(srn) { implicit request =>
+      (
+        for {
+          total <- request.userAnswers.get(TotalMemberContributionPage(srn, memberIndex, index)).getOrRecoverJourney
+          nameDOB <- request.userAnswers.get(MemberDetailsPage(srn, memberIndex)).getOrRecoverJourney
+        } yield Ok(
+          view(
+            form,
+            RemoveMemberContributionController.viewModel(
+              srn,
+              memberIndex: Max300,
+              index: Max50,
+              total,
+              nameDOB.fullName
+            )
+          )
+        )
+      ).merge
+    }
+
+  def onSubmit(srn: Srn, memberIndex: Max300, index: Max50): Action[AnyContent] =
+    identifyAndRequireData(srn).async { implicit request =>
+      form
+        .bindFromRequest()
+        .fold(
+          formWithErrors => {
+            (
+              for {
+                total <- request.userAnswers
+                  .get(TotalMemberContributionPage(srn, memberIndex, index))
+                  .getOrRecoverJourneyT
+                nameDOB <- request.userAnswers.get(MemberDetailsPage(srn, memberIndex)).getOrRecoverJourneyT
+              } yield BadRequest(
+                view(
+                  formWithErrors,
+                  RemoveMemberContributionController.viewModel(srn, memberIndex, index, total, nameDOB.fullName)
+                )
+              )
+            ).merge
+          },
+          removeDetails => {
+            if (removeDetails) {
+              for {
+                updatedAnswers <- Future
+                  .fromTry(request.userAnswers.remove(TotalMemberContributionPage(srn, memberIndex, index)))
```

Comment:
page 19, MemberContributionsPage (which is in the wrong package btw) should be removed too if we are removing the last contribution. If it remains, it will be a problem when posting data into ETMP. For reference please see: EmployerNamePage

Reply:
I am moving MemberContributionsPage into the `membercontributions` package